R/resolve_delayed.R

```diff
@@ -66,38 +68,50 @@
 #'
 #'   resolve_delayed(arm_ref_comp, ds)
 #' })
-resolve_delayed <- function(x, datasets) {
-  stopifnot(inherits(datasets, "FilteredData"))
+resolve_delayed <- function(x, datasets, keys = NULL) {
+  checkmate::assert(
+    combine = "and",
+    checkmate::check_list(datasets, type = "data.frame", min.len = 1),
+    !is.null(names(datasets)),
+    all(!names(datasets) %in% "")
+  )
```

Comment:
covers all scenarios

```
setNames(list(1, 2), c("a", "")) |> checkmate::check_list(names = "named")
list(1, 2) |> checkmate::check_list(names = "named")
setNames(list(1, 2), c("", "")) |> checkmate::check_list(names = "named")

```
```suggestion

checkmate::check_list(datasets, type = "data.frame", min.len = 1, names = "named)
```

Reply:
fixed